cmd/restic/cmd_ls.go

```diff
@@ -61,8 +65,81 @@ func runLs(opts LsOptions, gopts GlobalOptions, args []string) error {
 
 	ctx, cancel := context.WithCancel(gopts.ctx)
 	defer cancel()
+
+	return printLS(ctx, repo, gopts, opts, args)
+}
+
+type lsSnapshot struct {
+	*restic.Snapshot
+
+	ID      *restic.ID `json:"id"`
+	ShortID string     `json:"short_id"`
+	Nodes   []lsNode   `json:"nodes"`
+}
```

Comment:
Since the JSON structures that get printed are different, how about a field that indicates the type of struct it is, which is common between both of them? This way, we can easily differentiate the two in code.

Reply:
No, I mean a field like this:

```
Type string `json:"type"`
```

With a value of either "snapshot" or "node" so we know what kind of structure to parse out when we decode it.

Reply:
I guess we can but if I may... (sorry I like to understand)
- we already have value type on node for the kind of (file, dir...)
- why the need differentiate the two  (snaphost or node)  in code using a type value ? A Snapshot has a value named "nodes" which is an array of nodes and only nodes. isn't it redundant ?

Reply:
I guess that's a good point -- in that case, it should be documented how to distinguish the two. I still would prefer a separate field to indicate how to decode the structure, because it makes it so easy then to know how to decode it, rather than using a heuristic like looking for a certain field that only one struct has and the other doesn't. Maybe it could be called `StructType`, or the Type field in the lsNode struct could be renamed to FileType or ItemType or something.

Latest changes look great though!